main.rb

```diff
@@ -0,0 +1,199 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr = to_a if instance_of?(Range)
+    arr = flatten if instance_of?(Hash)
+
+    counter = 0
+    while counter < arr.length
+      yield(arr[counter])
+      counter += 1
+    end
+    arr
+  end
```

Comment:
- my_each when a block is given when self is a range returns the range itself after calling the given block once for each element in self
- my_each when a block is given when self is a hash behaves like #each calls the given block once for each element in self